File: examples/canonical-auth/java/src/main/java/dev/tenantcore/auth/routes/AuthController.java
package dev.tenantcore.auth.routes;

import dev.tenantcore.auth.auth.JwtService;
import dev.tenantcore.auth.domain.InMemoryStore;
import dev.tenantcore.auth.domain.User;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
public class AuthController {

    private final JwtService jwtService;
    private final InMemoryStore store;

    public AuthController(JwtService jwtService, InMemoryStore store) {
        this.jwtService = jwtService;
        this.store = store;
    }

    public record TokenRequest(String email, String password) {}

    @PostMapping("/auth/token")
    public ResponseEntity<?> token(@RequestBody TokenRequest req) {
        User user = store.getUserByEmail(req.email());
        if (user == null || !"password".equals(req.password())) {
            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
        }

        String token = jwtService.issueToken(user, store);
        return ResponseEntity.ok(Map.of(
            "access_token", token,
            "token_type", "Bearer",
            "expires_in", 900
        ));
    }
}

File: examples/canonical-auth/elixir/lib/tenantcore_auth/controllers/auth_controller.ex
defmodule TenantcoreAuth.AuthController do
  use TenantcoreAuth, :controller

  alias TenantcoreAuth.Auth.TokenService
  alias TenantcoreAuth.Domain.InMemoryStore

  def token(conn, %{"email" => email, "password" => "password"}) do
    case InMemoryStore.get_user_by_email(email) do
      nil -> conn |> put_status(:unauthorized) |> json(%{error: "Unauthorized"})
      user -> json(conn, %{access_token: TokenService.issue_token(user, InMemoryStore), token_type: "Bearer"})
    end
  end

  def token(conn, _params) do
    conn |> put_status(:unauthorized) |> json(%{error: "Unauthorized"})
  end
end
